tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -927,9 +927,7 @@ void Matmul::validate(
                         uint32_t per_core_M = program_config.per_core_M;
                         auto shard_shape = input_tensor_a.shard_spec().value().shape;
 
-                        // No padding
-                        TT_FATAL(M % per_core_M == 0);
-                        TT_FATAL(M / per_core_M == input_tensor_a.shard_spec().value().grid.num_cores());
+                        TT_FATAL(div_up(M, per_core_M) == input_tensor_a.shard_spec().value().grid.num_cores());
```

Comment:
Did you try out a few cases for padded? Does it work?

Reply:
Yes, I've added 1 unit test in test_sharded.py, and another test which represents falcon7b attention matmuls where we are hitting this issue in the first place if we try sharding. (this is in test_falcon_matmuls file)